libs/ui-watt/src/lib/components/shell/nav-list/watt-expand-on-active-link.directive.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Directive, AfterViewInit, Input, QueryList } from '@angular/core';
+import { MatExpansionPanel } from '@angular/material/expansion';
+import { from, mergeAll, filter } from 'rxjs';
+
+import { WattNavListItemComponent } from './watt-nav-list-item.component';
+
+@Directive({
+  selector: '[wattExpandOnActiveLink]',
+  exportAs: 'wattExpandOnActiveLink',
+  standalone: true,
+})
+export class WattExpandOnActiveLinkDirective implements AfterViewInit {
+  @Input()
+  wattNavListItemComponents: QueryList<WattNavListItemComponent> | null = null;
+
+  constructor(private panel: MatExpansionPanel) {}
+
+  ngAfterViewInit(): void {
+    const navListItems = this.wattNavListItemComponents?.toArray();
+
+    if (navListItems) {
+      const links$ = navListItems.map((item) => item.isActive$);
+
+      from(links$)
+        .pipe(
+          mergeAll(),
+          filter((isActive) => isActive)
+        )
+        .subscribe(() => {
+          this.panel.open();
+        });
+    }
```

Comment:
nitpick: perhaps combine the `map` and `mergeAll` into the `mergeMap` operator?
```ts
from(navListItems)
  .pipe(
    mergeMap(item => item.isActive$),
    filter((isActive) => isActive),
  )
```

Reply:
Magic 🔮 Nice snippet 🚀